### PR TITLE
Remove Immutable.js

### PR DESCRIPTION
**Pull request by fiskus**
I mostly replicated Immutable-specific code with plain javascript or Ramda. As I see frontend is working, but, maybe, I broke something

### CODE REVIEW

File: catalog/app/containers/Auth/SSOAzure.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { push } from 'connected-react-router/esm/immutable'
+import { push } from 'connected-react-router'
```
Comment (nl0) [inline catalog/app/containers/Auth/SSOAzure.js:1]: please backlog ditching `connected-react-router` as well

File: catalog/app/utils/Store.js [hunk 2]
```diff
@@ -25,21 +24,18 @@ export const Provider = function StoreProvider({ initialState = {}, history, chi
       process.env.NODE_ENV === 'development' &&
       process.env.LOGGER_REDUX === 'enabled'
     ) {
-      const stateTransformer = (state) =>
-        // pure JS is easier to read than Immutable objects
-        Iterable.isIterable(state) ? state.toJS() : state
       // eslint-disable-next-line global-require
       const { createLogger } = require('redux-logger')
-      middlewares.push(createLogger({ stateTransformer, collapsed: true }))
+      middlewares.push(createLogger({ stateTransformer: R.identity, collapsed: true }))
```
Comment (nl0) [inline catalog/app/utils/Store.js:29]: do we still need `stateTransformer` parameter here?

File: catalog/app/utils/ResourceCache.js [hunk 3]
```diff
@@ -57,83 +56,71 @@ const Action = tagged([
   'CleanUp', // { time: Date }
 ])
 
-const keyFor = (resource, input) => [resource.id, I.fromJS(resource.key(input))]
+const keyFor = (resource, input) => [resource.id, resource.key(input)]
```
Comment (nl0) [inline catalog/app/utils/ResourceCache.js:59]: i think this is where things have been broken: `resource.key` doesn't always return string, in fact, it usually returns some object passed as an input, and this object gets used as a key for the entry, so we're basically doing smth like `cache[{ some: input }] = entry`, which turns `cache` var into `{ '[object Object]': entry }` (check redux logs and inspect the state to make sure), so, it kinda works when there's no conflicts for every resource (only single item is fetched) or every key is a string, but the logic is broken.
solution: use some kind of proper map implementation which can use arbitrary objects as keys (js native `Map` is okayish, but it's mutable, and i'm not sure about its performance when using it as immutable -- creating a new map on every change)

**Review by nl0 — CHANGES_REQUESTED** [review]
ResourceCache is broken, we need to fix it

**Comment by nl0** [comment]
another option would be to allow only string keys for resources and explicitly require key construction functions for every resource (grep for `createResource` -- there's not that many resources defined currently)

File: catalog/app/containers/Auth/SSOAzure.js (hunk 1 above)
Comment (fiskus) [reply catalog/app/containers/Auth/SSOAzure.js:1]: I don't need task for this

File: catalog/app/utils/Store.js (hunk 2 above)
Comment (fiskus) [reply catalog/app/utils/Store.js:29]: I don't know. And I don't refactor code in this PR, only remove Immutable.js

Comment (nl0) [reply catalog/app/utils/Store.js:29]: well, you can remove it to avoid future confusion as to why was it kept

**Comment by nl0** [comment]
what's the status here? it seems ready, but abandoned

File: catalog/app/utils/Store.js (hunk 2 above)
Comment (fiskus) [reply catalog/app/utils/Store.js:29]: I shouldn't but I can remove it if you would be more polite

**Comment by fiskus** [comment]
It's not abandoned, I just have tasks with higher priority

**Pull request closed without merging**